Linker errors with --enable-executable-dynamic

Citing from http://hydra.cryp.to/build/1199867/nixlog/1/raw:

~~~
Linking dist/build/test-all/test-all ...
/tmp/nix-build-hpc-coveralls-0.9.0.drv-0/hpc-coveralls-0.9.0/dist/build/libHShpc-coveralls-0.9.0-6Q8vPLE5qvd6iS3fdA3Ztx-ghc7.10.2.so: undefined reference to `hpccozu6Q8vPLE5qvd6iS3fdA3ZZtx_Pathszuhpczucoveralls_version1_closure'
collect2: error: ld returned 1 exit status
~~~

This error goes away when I remove `--enable-executable-dynamic` from the set of configure flags.

--- a/hpc-coveralls.cabal
+++ b/hpc-coveralls.cabal
@@ -49,6 +49,7 @@ library
     Trace.Hpc.Coveralls.Util
   other-modules:
     HpcCoverallsCmdLine,
+    Paths_hpc_coveralls,
     Trace.Hpc.Coveralls.Cabal,
     Trace.Hpc.Coveralls.Config,
     Trace.Hpc.Coveralls.Curl,

